crates/uv-build-frontend/src/error.rs

```diff
@@ -192,24 +193,57 @@ impl Display for MissingHeaderCause {
                 {
                     write!(
                         f,
-                        "This error likely indicates that `{}` depends on `{}`, but doesn't declare it as a build dependency. If `{}` is a first-party package, consider adding `{}` to its `{}`. Otherwise, `{}` into the environment and re-run with `{}`.",
+                        "This error likely indicates that `{}` depends on `{}`, but doesn't declare it as a build dependency. \
+                        If `{}` is a first-party package, consider adding `{}` to its `{}`. \
+                        Otherwise, either add it to your `pyproject.toml` under:\n\
+                        \n\
+                            [tool.uv.extra-build-dependencies]\n\
+                            \"{}\" = [\"{}\"]\n\
+                        \n\
+                        or `{}` into the environment and re-run with `{}`.",
                         format!("{package_name}@{package_version}").cyan(),
                         package.cyan(),
                         package_name.cyan(),
                         package.cyan(),
                         "build-system.requires".green(),
+                        package_name.cyan(),
+                        package.cyan(),
                         format!("uv pip install {package}").green(),
                         "--no-build-isolation".green(),
                     )
                 } else if let Some(version_id) = &self.version_id {
+                    // https://peps.python.org/pep-0508/#names
+                    // ^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])$ with re.IGNORECASE
+                    // Since we're only using this for a hint, we're more lenient than what we would be doing if this was used for parsing
+                    let end = version_id
+                        .char_indices()
+                        .take_while(|(_, char)| matches!(char, 'A'..='Z' | 'a'..='z' | '0'..='9' | '.' | '-' | '_'))
+                        .last()
+                        .map_or(0, |(i, c)| i + c.len_utf8());
+
+                    let normalized_version_id = if end == 0 {
+                        version_id
+                    } else {
+                        &version_id[..end]
+                    };
```

Comment:
Maybe we should split this into a utility function to make it clear what we're doing. I'd also probably call the variable `package_name` instead of `normalized_version_id`